File: echolot-demo/src/main/java/de/exxcellent/echolot/testapp/testscreen/PlainHtmlTest.java
/*
 * This file (PlainHtmlTest.java) is part of the Echolot Project (hereinafter "Echolot").
 * Copyright (C) 2008-2010 eXXcellent Solutions GmbH.
 *
 * Version: MPL 1.1/GPL 2.0/LGPL 2.1
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 1.1 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * Alternatively, the contents of this file may be used under the terms of
 * either the GNU General Public License Version 2 or later (the "GPL"), or
 * the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
 * in which case the provisions of the GPL or the LGPL are applicable instead
 * of those above. If you wish to allow use of your version of this file only
 * under the terms of either the GPL or the LGPL, and not to allow others to
 * use your version of this file under the terms of the MPL, indicate your
 * decision by deleting the provisions above and replace them with the notice
 * and other provisions required by the GPL or the LGPL. If you do not delete
 * the provisions above, a recipient may use your version of this file under
 * the terms of any one of the MPL, the GPL or the LGPL.
 */

package de.exxcellent.echolot.testapp.testscreen;

import de.exxcellent.echolot.app.PlainHtml;
import de.exxcellent.echolot.testapp.ButtonColumn;
import nextapp.echo.app.Extent;
import nextapp.echo.app.Label;
import nextapp.echo.app.SplitPane;
import nextapp.echo.app.event.ActionEvent;
import nextapp.echo.app.event.ActionListener;

/**
 * Simple Test-Class for the PlainHtml-Component
 * @author Ralf Enderle
 * @version 1.0
 */
public class PlainHtmlTest extends SplitPane {

    private static final String HTML__SOME_TEXT = "Just some simple text with one <b>bold</b> item";
    private static final String HTML__TABLE = "" +
            "<table border=\"1\">  " +
            "<tr><th>Stadt</th> <th>Land</th> <th>Fluss</th> </tr>  " +
            "<tr> <td>Paris</td> <td>Frankreich</td> <td>Seine</td> </tr>" +
            "<tr> <td>Frankfurt</td> <td>Deutschland</td> <td>Main</td> </tr>" +
            "</table>  ";
    private static final String HTML__YOUTUBE = "" +
            "<iframe class=\"youtube-player\" type=\"text/html\" width=\"640\" height=\"385\" " +
            "src=\"http://www.youtube.com/embed/9_xl9boG8Qg\" frameborder=\"0\"> </iframe>";

    public PlainHtmlTest() {
        super(SplitPane.ORIENTATION_HORIZONTAL, new Extent(250, Extent.PX));
        setStyleName("DefaultResizable");

        ButtonColumn controlsColumn = new ButtonColumn();
        controlsColumn.setStyleName("TestControlsColumn");
        add(controlsColumn);

        final PlainHtml plainHtml = new PlainHtml("Hi there :-)");
        //plainHtml.setHtmlText("Test");

        add(plainHtml);

        controlsColumn.add(new Label("Tooltip shows the html that will be displayed!"));
        controlsColumn.addButton("Just some text", HTML__SOME_TEXT, new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                plainHtml.setHtmlText(HTML__SOME_TEXT);
            }
        });

        controlsColumn.addButton("Let's show a table", HTML__TABLE, new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                plainHtml.setHtmlText(HTML__TABLE);
            }
        });

        controlsColumn.addButton("Let's show an embedded Youtube iFrame", HTML__YOUTUBE, new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                plainHtml.setHtmlText(HTML__YOUTUBE);
            }
        });
    }

}

File: test/testscreen/PlainHtmlTest.js
/*
 * This file (PlainHtmlTest.js) is part of the Echolot Project (hereinafter "Echolot").
 * Copyright (C) 2008-2011 eXXcellent Solutions GmbH.
 *
 * Version: MPL 1.1/GPL 2.0/LGPL 2.1
 */

exxcellent.test.PlainHtmlTest = Core.extend({

    _plainHtml : null,

    $construct: function(testArea) {
        var testColumn = new Echo.Column({styleName: "Default"});
        var controlColumn = new Echo.Column({
          styleName: "TestControl",
          children: [
               new Echo.Label({
                   text: "PlainHtml", styleName: "Title"
               }),
               new Echo.Label({
                   text: "Content", styleName: "Default"
               }),
               this._createButton("Default-Text", "Just a text"),
               this._createButton("Liste","<ul><li>Item 1</li><li>Item 2</li>" +
                        "<p>" +
                        "<b>Und noch was Fett</b>"),
               this._createButton('Show Table','<table border="1">  ' +
                        '<tr><th>Stadt</th> <th>Land</th> <th>Fluss</th> </tr>  ' +
                        '<tr> <td>Paris</td> <td>Frankreich</td> <td>Seine</td> </tr>' +
                        '<tr> <td>Frankfurt</td> <td>Deutschland</td> <td>Main</td> </tr>' +
                        '</table>  '),
               this._createButton("Youtube iFrame",
                '<iframe class="youtube-player" type="text/html" width="640" height="385" src="http://www.youtube.com/embed/9_xl9boG8Qg" frameborder="0"> </iframe>')
          ]
        });

        this._plainHtml = new exxcellent.PlainHtml({
            htmlText : "Just a text"
        });

        testColumn.add(this._plainHtml);

        testArea.add(new Echo.SplitPane({
            styleName: "TestControl",
            children: [
                controlColumn,
                testColumn]
        }));
    },
    /**
     * Helper
     * @param text
     * @param action
     */
    _createButton: function(text, anotherHtml) {
        return new Echo.Button({
            text: text,
            styleName: "Default",
            toolTipText: anotherHtml,
            events: {
                action: Core.method(
                        this, this._setAnotherHtml(anotherHtml))
            }
        });
    },

    _setAnotherHtml : function(anotherHtml) {
        return function() {
            this._plainHtml.set(exxcellent.PlainHtml.HTML_TEXT, anotherHtml);
        }
    }
});